opta/utils/runtee.py

```diff
@@ -0,0 +1,159 @@
+# type: ignore
+# nosec
+"""tee like run implementation."""
+import asyncio
+import os
+import platform
+import subprocess  # nosec
+import sys
+from asyncio import StreamReader
+from typing import TYPE_CHECKING, Any, Callable, Dict, List, Optional, Union
+
+if TYPE_CHECKING:
+    CompletedProcess = subprocess.CompletedProcess[Any]  # pylint: disable=E1136
+else:
+    CompletedProcess = subprocess.CompletedProcess
+
+try:
+    from shlex import join  # type: ignore
+except ImportError:
+    from subprocess import list2cmdline as join  # nosec
```

Comment:
```suggestion
from shlex import join
```

`shlex.join` is part of the standard library in Python 3.8, so there shouldn't be a case where it's not present.